jekyll/_cci2/testing-orbs.md

```diff
@@ -0,0 +1,175 @@
+
+The following steps illustrate doing basic expansion testing from the CLI:
+
+1) Assume a simple orb in `src/orb.yml`
+
+{% raw %}
+```
+version: 2.1
+
+executors:
+  default:
+    parameters:
+      tag:
+        type: string
+        default: "curl-browsers"
+      docker:
+        - image:  circleci/buildpack-deps:parameters.tag
+
+jobs:
+  hello-build:
+    executor: default
+    steps:
+      - run: echo "Hello, build!"
+```
+{% endraw %}
+
+2) Validate the orb with `circleci orb validate src/orb.yml`.
+
+3) Publish a dev version with `circleci orb publish src/orb.yml namespace/orb@dev:0.0.1`.
+
+4) Include that orb in `.circleci/config.yml`:
+
+{% raw %}
+```
+version: 2.1
+
+orbs:
+  hello: namespace/orb@dev:0.0.1
+
+workflows:
+  hello-workflow:
+    jobs:
+      - hello/hello-build
+```
+{% endraw %}
+
+After running `circleci config process .circleci/config.yml` the expected result would be:
+
+{% raw %}
+```
+version: 2
+jobs:
+  hello/hello-build:
+    docker:
+      - image: circleci/buildpack-deps:curl-browsers
+    steps:
+      - run:
+          command: echo "Hello, build!"
+workflows:
+  hello-workflow:
+    jobs:
+      - hello/hello-build
+  version: 2
+```
+{% endraw %}
+
+The `config.yml` file should look like the following:
+
+{% raw %}
+```
+version: 2.1
+
+orbs:
+  hello: namespace/orb@dev:0.0.1
+
+  workflows:
+    hello-workflow:
+    jobs:
+        - hello/hello-build
+```
+{% endraw %}
+
+The result shown above may now be used in a custom script that tests for its structure relative to what is expected. This form of testing is useful for ensuring you do not break any contracts your orb interface has established with users of the orb, while also enabling you to test different parameter inputs to determine how these parameter imputs impact what is generated during config processing.
+
+### Runtime Testing
+
+Runtime testing involves running active builds with orbs. Because the jobs in a build can only depend on orbs that are either part of the config or were published when the build started, this technique requires some special planning.
+
+One approach is to run jobs within your build's jobs by using the circleci CLI to run local builds using `circleci local execute` on a machine executor within your builds. This allows you to print the build output to `stdout` so you can make assertions about it. This approach can be limiting, however, because local builds do not support workflows and have other caveats. This is also powerful if you need to test the actual running output of a build using your orb elements. 
+
+For an example of using this technique see the [Artifactory Orb](https://github.com/CircleCI-Public/artifactory-orb) page in CircleCI's public GitHub page.
```

Comment:
change CircleCI's to avoid using possessive

Reply:
Fixed